sparse/coo/core.py

```diff
@@ -278,6 +280,9 @@ def from_numpy(cls, x):
         """
         x = np.asanyarray(x)
         if x.shape:
+            if approx and x.dtype and x.dtype.kind == 'f':
+                i = np.abs(x) < np.finfo(x.dtype).resolution * 10
```

Comment:
I'd compare the value with the arithmetic mean of the entire array multiplied by `np.finfo(x.dtype).resolution * 10`. We don't necessarily know the range of the inputs here.